### DISCUSSION TITLE
How to create the Table of contents?

### DISCUSSION THREAD

**Discussion by TonikRU** [ONLYOFFICE/DocumentBuilder#90] [Q&A] [answered]
Hello!

Is it possible to create the table of contents of a document using DocumentBuilder API?
I mean this:
![table of contents](https://user-images.githubusercontent.com/7153705/184859065-1dbaf0d1-11fd-478d-8147-469229b706bb.png)

DocumentBuilder version: 7.1
OS: Windows

Thank you.

**Comment by l8556** [comment]
Hi @TonikRU !
These methods in development.
Improvement request in our internal repository #58561

**Comment by trofim24** [comment]
We've implemented in API [addtableofcontents](https://api.onlyoffice.com/docbuilder/textdocumentapi/apidocument/addtableofcontents) method in version 7.3.0

**Comment by trofim24** [comment, marked as answer]
You can use `AddTableOfContents` [method](https://api.onlyoffice.com/docs/office-api/usage-api/text-document-api/ApiDocument/Methods/AddTableOfContents/)
